CRM/MembershipExtras/Hook/ValidateForm/MembershipType.php

```diff
@@ -45,6 +45,10 @@ public function validate() {
       $this->errors['duration_unit'] = ts('Fixed period membership type only supports 1 year duration.');
     }
 
+    if (empty($this->fields['membership_type_annual_pro_rata_calculation'])) {
```

Comment:
I would move this too into separate private function and add dockblock explaining what it validates.